uaclient/actions.py

```diff
@@ -268,6 +266,11 @@ def collect_logs(cfg: config.UAConfig, output_dir: str):
             "{}/{}.txt".format(output_dir, service),
             return_codes=[0, 3],
         )
+    pro_status, _ = status(cfg=cfg, show_all=False)
```

Comment:
That works, but when calling the `pro status --format json`, there are some extra steps performed on the status dict before converting it to JSON:
https://github.com/canonical/ubuntu-pro-client/blob/main/uaclient/event_logger.py#L44

Another difference is that `pro status --format json` will always produce a JSON response, even if it errors out. However, I think calling the `_write_command_output_to_file` and directly writing the exception that was raised is a better approach.

But I am unsure for the first point I raised. @orndorffgrant @renanrodrigo do you think we should output the exact same JSON structure as `pro status --format json` ?

Reply:
I don't think we need to keep the exact format, since these are just debug artifacts for us to look at.

It does seem that function adds the pro environment variables, so the current PR loses that information.

@dheyay I think we can add a separate artifact that also saves the output of `util.get_pro_environment` in a file

Reply:
Sure, I'll take a look at that